src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -68,24 +72,185 @@ internal class ResourceOperation : TypeProvider
+                    !pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/resourceGroups/"))
+                {
+                    return listMethod;
+                }
+            }
+
+            return null;
+        }
+
+        public RestClientMethod? GetRestClientMethod(ResourceListMethod? listMethod)
+        {
+            if (listMethod != null)
+            {
+                if (listMethod.PagingMethod != null)
+                {
+                    return listMethod.PagingMethod.Method;
+                }
+                else if (listMethod.ClientMethod != null)
+                {
+                    return listMethod.ClientMethod.RestClientMethod;
+                }
+            }
+
+            return null;
+        }
+
+        protected IEnumerable<ResourceListMethod> GetListMethods(bool parentExistsInPathParam, bool returnTypeIsResourceData)
+        {
+            List<ResourceListMethod> listMethods = new List<ResourceListMethod>();
+            var parentResourceType = OperationGroup.ParentResourceType(_context.Configuration.MgmtConfiguration);
+
+            foreach (var pagingMethod in PagingMethods)
+            {
+                if (IsValidListMethod(parentExistsInPathParam, returnTypeIsResourceData, pagingMethod.Method))
+                {
+                    listMethods.Add(new ResourceListMethod(pagingMethod, null));
+                }
+            }
+
+            foreach (var method in Methods)
+            {
+                if (IsValidListMethod(parentExistsInPathParam, returnTypeIsResourceData, method.RestClientMethod))
+                {
+                    listMethods.Add(new ResourceListMethod(null, method));
+                }
+            }
+
+            return listMethods;
+        }
+
+        private bool IsValidListMethod(bool parentExistsInPathParam, bool returnTypeIsResourceData, RestClientMethod clientMethod)
+        {
+            var parentResourceType = OperationGroup.ParentResourceType(_context.Configuration.MgmtConfiguration);
+            bool isParentExistsInPathParams = false;
+            bool isReturnTypeResourceData = false;
+
+            var parentParamName = "";
+            if (clientMethod.Operation?.Requests.FirstOrDefault().Protocol.Http is HttpRequest httpRequest)
+            {
+                var path = httpRequest.Path;
+                var parentIndex = path.IndexOf(parentResourceType);
+                if (parentIndex > 0)
+                {
+                    path = path.Substring(parentIndex + parentResourceType.Length);
+                    var pathArr = path.Split('/');
+                    char[] charsToTrim = { '{', '}' };
+                    parentParamName = pathArr[1].Trim(charsToTrim);
+
+                    isParentExistsInPathParams = clientMethod.PathParameters.Any(p => p.Name == parentParamName);
```

Comment:
Based on the "we have to trim the `{` and `}`" logic here, I assume the answer to my above question should be NO - then we might never get the correct parent other than the parent is `subscriptions`, `resourceGroups` or first level resources.
The string of sub resource type (e.g. `Microsoft.Network/virtualNetworks/subnets`) will never show in the request path.

Also it would help tons if we have some utility functions to do this pattern, I feel like we also need this when we are trying to get the correct parameters to pass in the operation writer or container writer